Xamarin.Forms.Platform.Android/InnerGestureListener.cs

```diff
@@ -4,222 +4,236 @@
+        {
+            SetStartingPosition(e);
+        }
+
+        bool GestureDetector.IOnGestureListener.OnScroll(MotionEvent e1, MotionEvent e2, float distanceX, float distanceY)
+        {
+            if (e1 == null || e2 == null)
+                return false;
+
+            SetStartingPosition(e1);
+
+            return StartScrolling(e2);
+        }
+
+        void GestureDetector.IOnGestureListener.OnShowPress(MotionEvent e)
+        {
+        }
+
+        bool GestureDetector.IOnGestureListener.OnSingleTapUp(MotionEvent e)
+        {
+            if (_disposed)
+                return false;
+
+            if (HasDoubleTapHandler())
+            {
+                // Because we have a handler for double-tap, we need to wait for
+                // OnSingleTapConfirmed (to verify it's really just a single tap) before running the delegate
+                return false;
+            }
+
+            // A single tap has occurred and there's no handler for double tap to worry about,
+            // so we can go ahead and run the delegate
+            return _tapDelegate(1);
+        }
+
+        bool GestureDetector.IOnDoubleTapListener.OnSingleTapConfirmed(MotionEvent e)
+        {
+            if (_disposed)
+                return false;
+
+            if (!HasDoubleTapHandler())
+            {
+                // We're not worried about double-tap, so OnSingleTapUp has already run the delegate
+                // there's nothing for us to do here
+                return false;
+            }
+
+            // Since there was a double-tap handler, we had to wait for OnSingleTapConfirmed;
+            // Now that we're sure it's a single tap, we can run the delegate
+            return _tapDelegate(1);
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            _disposed = true;
+
+            if (disposing)
+            {
+                _tapDelegate = null;
+                _tapGestureRecognizers = null;
+                _scrollDelegate = null;
+                _scrollStartedDelegate = null;
+                _scrollCompleteDelegate = null;
+                _swipeDelegate = null;
+                _swipeCompletedDelegate = null;
+            }
+
+            base.Dispose(disposing);
+        }
+
+        void SetStartingPosition(MotionEvent e1)
+        {
+            _lastX = e1.GetX();
+            _lastY = e1.GetY();
+        }
+
+        bool StartScrolling(MotionEvent e2)
+        {
+            if (_scrollDelegate == null)
+                return false;
+
+            if (!_isScrolling && _scrollStartedDelegate != null)
+                _scrollStartedDelegate(e2.PointerCount);
+
+            _isScrolling = true;
+
+            float totalX = e2.GetX() - _lastX;
+            float totalY = e2.GetY() - _lastY;
+
+            return _scrollDelegate(totalX, totalY, e2.PointerCount) || _swipeDelegate(totalX, totalY);
```

Comment:
When a PanGestureRecognizer was added, SwipeGestureRecognizer was not working, 
Is it intended?